Framework/CMakeLists.txt

```diff
@@ -57,6 +57,10 @@ if ( ${CMAKE_SYSTEM_NAME} MATCHES "Linux" )
   endif ( MPI_BUILD )
 endif ()
 
+if ( MPI_EXPERIMENTAL )
```

Comment:
It has nothing to do with algorithms. My intention is to merge it with the existing SNS MPI build, which is named `MPI_BUILD`, but as explained in the PR I cannot do that right now.